lib/dao.js

```diff
@@ -1610,9 +1610,12 @@ DataAccessObject._coerce = function(where, props) {
       if (props[model]) {
         var clause = {};
         clause[prop] = where[p];
-        where[p] = Array.isArray(props[model].type) ?
-          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
-          self._coerce(clause, props[model].type.definition.properties)[prop];
+        var type = Array.isArray(props[model].type) ? props[model].type[0] : props[model].type;
+        if (type && type.definition && type.definition.properties) {
+          where[p] = self._coerce(clause, type.definition.properties)[prop];
+          continue;
```

Comment:
Minor nitpick: this `continue` call is unnecessary, as it will fall through to the next one.